meta/autoid/autoid_test.go

```diff
@@ -120,3 +122,69 @@ func (*testSuite) TestT(c *C) {
 	c.Assert(err, IsNil)
 	c.Assert(id, Equals, int64(6544))
 }
+
+func (*testSuite) TestAsyn(c *C) {
+	driver := localstore.Driver{Driver: goleveldb.MemoryDriver{}}
+	store, err := driver.Open("memory")
+	c.Assert(err, IsNil)
+	defer store.Close()
+	step = 100
+	defer func() {
+		step = 5000
+	}()
+
+	dbID := int64(2)
+	tblID := int64(100)
+	err = kv.RunInNewTxn(store, false, func(txn kv.Transaction) error {
+		m := meta.NewMeta(txn)
+		err = m.CreateDatabase(&model.DBInfo{ID: dbID, Name: model.NewCIStr("a")})
+		c.Assert(err, IsNil)
+		err = m.CreateTable(dbID, &model.TableInfo{ID: tblID, Name: model.NewCIStr("t")})
+		c.Assert(err, IsNil)
+		return nil
+	})
+	c.Assert(err, IsNil)
+
+	var mu sync.Mutex
+	wg := sync.WaitGroup{}
+	m := map[int64]struct{}{}
+	errCh := make(chan error, 100)
+
+	allocIDs := func() {
+		alloc := NewAllocator(store, dbID)
+		for j := 0; j < int(step)+5; j++ {
+			id, err := alloc.Alloc(tblID)
+			if err != nil {
+				errCh <- err
+				break
+			}
+
+			mu.Lock()
+			if _, ok := m[id]; ok {
+				errCh <- fmt.Errorf("duplicate id:%v", id)
+				mu.Unlock()
+				break
+			}
+			m[id] = struct{}{}
+			mu.Unlock()
+		}
+	}
+	for i := 0; i < 100; i++ {
+		wg.Add(1)
+		time.Sleep(time.Duration(i%10) * time.Microsecond)
```

Comment:
Make some allocators start at the same time, some allocators don't start at the same time.